File: Java/Control Statements/NthFibonacciNumber.java

public class NthFibonacciNumber {
    public static int fibonacciNumber(int n) {
        long mod = 1000000007;
        long a = 0;
        long b = 1;
        long c = 0;
        for (int i = 0; i < n; i++) {
            c = (a % mod + b % mod);
            a = b % mod;
            b = c % mod;
        }
        return (int) a;
    }
}

File: Python/Conditional statements & loop/N-th Fibonacci Number.py
l = []
l.append(0)
l.append(1)


def fibonacciNumber(n):
    MOD = 1000000007
    if (n == 1):
        return n
    if (n < len(l)):
        return l[n] % MOD
    for i in range(len(l), n+1):
        l.append((l[i-1] % MOD + l[i-2] % MOD) % MOD)
    return l[n]
